Tools/jit/jit.h

```diff
@@ -0,0 +1,3 @@
+// To use preserve_none in JIT builds, we need to declare a separate function pointer with __attribute__((preserve_none)) since this attribute is not supported in < clang 19.
+// This is functionally the same as jit_func_native from Include/internal/pycore_jit.h except that it has __attribute__((preserve_none)).
+typedef _Py_CODEUNIT *(*jit_func_preserve_none)(_PyInterpreterFrame *frame, _PyStackRef *stack_pointer, PyThreadState *tstate) __attribute__((preserve_none));
```

Comment:
I think this works, and it means we don't need to keep the two signatures in sync! It basically says "`jit_func_preserve_none` is `jit_func`, but with `__attribute__((preserve_none))`":

```suggestion
typedef jit_func __attribute__((preserve_none)) jit_func_preserve_none;
```